src/datasets/iterable_dataset.py

```diff
@@ -1218,13 +1265,19 @@ def __init__(
 
     @property
     def iter_arrow(self):
-        if self.formatting and self.formatting.format_type == "arrow":
+        # if self.formatting is not None, we format, filter, then return arrow iterator
+        # if formatting is unset, we don't, as that would require filter fn to accept arrow format
+        if self.formatting and self.ex_iterable.iter_arrow:
```

Comment:
I think it's ok to not return iter_arrow if the formatting is not "arrow", this way we decode the data only once in the pipeline (e.g. if it contains image/audio data)